rpcserver.go

```diff
@@ -7825,9 +7848,40 @@ func (r *rpcServer) rfqChannel(ctx context.Context, id asset.ID,
 
 	// If the user specified a peer public key, and we still have multiple
 	// channels, it means we have multiple channels with the same asset and
-	// the same peer. So we just return the first channel.
+	// the same peer.
 	case len(assetBalances) >= 1:
-		return &assetBalances[0], nil
+		// Initialize to first element.
+		bestBalance := assetBalances[0]
+
+		switch intention {
+		case ReceiveIntention:
+			// If the intention is to receive, return the channel
+			// with the best remote balance.
+			fn.ForEach(assetBalances, func(b channelWithAsset) {
+				if b.assetInfo.RemoteBalance >
+					bestBalance.assetInfo.RemoteBalance {
+
+					bestBalance = b
+				}
+			})
+
+		case SendIntention:
+			// If the intention is to send, return the channel with
+			// the biggest local balance.
+			fn.ForEach(assetBalances, func(b channelWithAsset) {
```

Comment:
Just to check my understanding: When sending, this code path only really matters if the `peerPubKey` isn't set and you have multiple channels with a given asset to different peers. Then we select the peer with the best inbound liquidity to us (the actual channel is ignored on the send related call sites).
But if `peerPubKey` is specified, this is basically a no-op, since it doesn't matter which channel of a peer we choose, as we only need the pubKey anyway.